--- system.dts.orig	2024-12-17 02:53:50.728339591 +0000
+++ system.dts	2024-12-20 03:22:38.724008387 +0000
@@ -203,7 +203,7 @@
 
 	firmware {
 
-		zynqmp-firmware {
+		zynqmp_firmware: zynqmp-firmware {
 			compatible = "xlnx,zynqmp-firmware";
 			#power-domain-cells = <0x01>;
 			method = "smc";
@@ -715,7 +715,7 @@
 			phandle = <0x41>;
 		};
 
-		interrupt-controller@f9010000 {
+		gic: interrupt-controller@f9010000 {
 			compatible = "arm,gic-400";
 			#interrupt-cells = <0x03>;
 			reg = <0x00 0xf9010000 0x00 0x10000 0x00 0xf9020000 0x00 0x20000 0x00 0xf9040000 0x00 0x20000 0x00 0xf9060000 0x00 0x20000>;
@@ -1575,7 +1575,7 @@
 			pinctrl-names = "default";
 			u-boot,dm-pre-reloc;
 			compatible = "xlnx,zynqmp-uart\0cdns,uart-r1p12";
-			status = "okay";
+			status = "disabled";
 			interrupt-parent = <0x05>;
 			interrupts = <0x00 0x16 0x04>;
 			reg = <0x00 0xff010000 0x00 0x1000>;
@@ -1921,18 +1921,6 @@
 		reg = <0x00 0x00 0x00 0x80000000 0x08 0x00 0x00 0x80000000>;
 	};
 
-	reserved-memory {
-		#address-cells = <0x02>;
-		#size-cells = <0x02>;
-		ranges;
-
-		pmu@7ff00000 {
-			reg = <0x00 0x7ff00000 0x00 0x100000>;
-			no-map;
-			phandle = <0x7a>;
-		};
-	};
-
 	gpio-keys {
 		compatible = "gpio-keys";
 		autorepeat;
@@ -1973,6 +1961,68 @@
 		pwms = <0x1f 0x02 0x9c40 0x00>;
 	};
 
+       reserved-memory {
+           #address-cells = <2>;
+           #size-cells = <2>;
+           ranges;
+           rproc_0_reserved: rproc_0_reserved@3ec00000 {
+               no-map;
+               reg = <0x0 0x3ec00000 0x0 0x1000000>;
+	   };
+
+       };
+       tcm_0a: tcm_0a@ffe00000 {
+           no-map;
+           reg = <0x0 0xffe00000 0x0 0x10000>;
+           status = "okay";
+           compatible = "mmio-sram";
+           power-domain = <&zynqmp_firmware 15>;
+       };
+       tcm_0b: tcm_0b@ffe20000 {
+           no-map;
+           reg = <0x0 0xffe20000 0x0 0x10000>;
+           status = "okay";
+           compatible = "mmio-sram";
+           power-domain = <&zynqmp_firmware 16>;
+       };
+
+
+       rf5ss@ff9a0000 {
+           compatible = "xlnx,zynqmp-r5-remoteproc";
+           xlnx,cluster-mode = <1>;
+           ranges;
+           reg = <0x0 0xFF9A0000 0x0 0x15000>;
+           #address-cells = <0x2>;
+           #size-cells = <0x2>;
+           r5f_0 {
+               compatible = "xilinx,r5f";
+               #address-cells = <2>;
+               #size-cells = <2>;
+               ranges;
+               sram = <&tcm_0a &tcm_0b>;
+	       memory-region = <&rproc_0_reserved>;
+	       power-domain = <&zynqmp_firmware 7>;
+		};
+	};
+
+	amba_pl: amba_pl@0 {
+                #address-cells = <2>;
+                #size-cells = <2>;
+                compatible = "simple-bus";
+                ranges ;
+
+		shm0: shm@3ed20000 {
+			compatible = "shm_uio";
+			reg = <0x0 0x3ed20000 0x0 0x0140000>;
+		};
+		ipi0: ipi@ff340000 {
+			compatible = "ipi_uio";
+			reg = <0x0 0xff340000 0x0 0x1000>;
+			interrupt-parent = <&gic>;
+			interrupts = <0 29 4>;
+		};
+	};
+
 	__symbols__ {
 		pinctrl_usb1_default = "/firmware/zynqmp-firmware/pinctrl/usb1-default";
 		pinctrl_usb0_default = "/firmware/zynqmp-firmware/pinctrl/usb0-default";
